crypto/keybackup.go

```diff
@@ -0,0 +1,155 @@
+package crypto
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"maunium.net/go/mautrix/crypto/backup"
+	"maunium.net/go/mautrix/crypto/olm"
+	"maunium.net/go/mautrix/crypto/signatures"
+	"maunium.net/go/mautrix/id"
+)
+
+func (mach *OlmMachine) DownloadAndStoreLatestKeyBackup(ctx context.Context, megolmBackupKey *backup.MegolmBackupKey) error {
```

Comment:
This function should probably be split up a bit so we can have functions for downloading a specific session and maybe all sessions for a specific room

Reply:
I'll leave that for a future refactor once we actually want to do those things.